### ISSUE TITLE
Swift Compiler Error in Xcode 11.4

### ISSUE THREAD

**Issue by unmind-alex** [launchdarkly/react-native-client-sdk#44]
**Describe the bug**
Our project is failing to compile under Xcode 11.4, due to a Swift compile error within the auto-linked LaunchDarkly SDK.

![image](https://user-images.githubusercontent.com/50375793/77786164-3e6bba80-7055-11ea-8883-846e9af24566.png)

**To reproduce**
Attempt to compile a vanilla React Native project with the auto-linked LaunchDarkly SDK with Xcode 11.4.

**Expected behavior**
Successful compilation.

**Logs**
```
Protocol 'Collection' can only be used as a generic constraint because it has Self or associated type requirements
```

**SDK version**
<details>
<summary>JavaScript</summary>
<pre>
launchdarkly-js-client-sdk@2.17.0:
  version "2.17.0"
  resolved "https://registry.yarnpkg.com/launchdarkly-js-client-sdk/-/launchdarkly-js-client-sdk-2.17.0.tgz#7be9f1b12f7a0ea90ba9dfed73d076e97b78656a"
  integrity sha512-KLKFtud5k9tH2BsaCf7dNyfqaHM4MITLQbriDpqi5zYDAhNYh6qZRIuriCTOohhw0/0GoA+oMHDhI4zDQ7nVww==
  dependencies:
    escape-string-regexp "^1.0.5"
    launchdarkly-js-sdk-common "3.2.2"
launchdarkly-js-sdk-common@3.2.2:
  version "3.2.2"
  resolved "https://registry.yarnpkg.com/launchdarkly-js-sdk-common/-/launchdarkly-js-sdk-common-3.2.2.tgz#26f8a8bcf8e3391e3bf6fdf72e5883d3cc8a9c7d"
  integrity sha512-WBIMk3c1s3JGpSvDRZpDr0D4dDPsFkscDmcOoaoXnkW6PdacfUz9KKJ9bKTM+akNWE2DsZ3ttEz4rATEFeOnVA==
  dependencies:
    base64-js "^1.3.0"
    fast-deep-equal "^2.0.1"
    uuid "^3.3.2"
launchdarkly-react-client-sdk@^2.18.0:
  version "2.18.0"
  resolved "https://registry.yarnpkg.com/launchdarkly-react-client-sdk/-/launchdarkly-react-client-sdk-2.18.0.tgz#7563878ae9768ba227f2bdd6779f88dde1f7eebd"
  integrity sha512-g4jyNbDSL4qptnV9Hmca8b1Z/bXiF1GXVuEa6pISNSo4pdnPgKYjUshjiRIBq1hoi3+oO4PPQhS5DxdVSf6lVA==
  dependencies:
    launchdarkly-js-client-sdk "2.17.0"
    lodash.camelcase "^4.3.0"
    uuid "^3.3.2"
launchdarkly-react-native-client-sdk@^3.0.0:
  version "3.0.0"
  resolved "https://registry.yarnpkg.com/launchdarkly-react-native-client-sdk/-/launchdarkly-react-native-client-sdk-3.0.0.tgz#24a5a30544f215fd188a69a92183150387419db7"
  integrity sha512-gmeQpTwFoH6E3PNJL4nZv2fdIORoPRFnxt5+Ql6TLShvMcFNBxGaA/ZlGLOXW79zxaXd51Fw4F5xtIDXw8jQtA==
</pre>
</details>

<details>
<summary>CocoaPods</summary>
<pre>
  - LaunchDarkly (4.4.1):
    - LaunchDarkly/Core (= 4.4.1)
  - LaunchDarkly/Core (4.4.1):
    - DarklyEventSource (~> 4.1.0)
  - LaunchdarklyReactNativeClient (3.0.0):
    - LaunchDarkly (= 4.4.1)
</pre>
</details>

**Language version, developer tools**
Xcode 11.4 (11E146)

**OS/platform**
macOS 10.15.4

**Comment by torchhound** [comment]
Hi @unmind-alex, we have fixed this in the latest iOS SDK release. This change will be in a new version of the React Native SDK soon.

**Comment by bwoskow-ld** [comment]
Hi @unmind-alex ,

This issue has been fixed in version [3.0.2](https://github.com/launchdarkly/react-native-client-sdk/releases/tag/3.0.2) of the SDK.

Cheers,
Ben
